include/librfu.h

```diff
@@ -93,143 +101,204 @@ struct __attribute__((packed, aligned(2))) UnkLinkRfuStruct_02022B14
-    u8 block2[0x30];
+    union RfuPacket rxPacketAlloc;
+    union RfuPacket txPacketAlloc;
+    u8 block1[0x960]; // size of librfu_intr.s binary
+    struct RfuStruct block2;
 };
 
-struct RfuUnk1
+struct RfuSlotStatusUNI
 {
-    u16 unk_0;
-    u8 unk_2;
-    u8 unk_3;
-    u8 fill_4[14];
-    u8 unk_12;
-    u32 unk_14;
-    u32 unk_18;
-    struct RfuIntrStruct unk_1c;
+    struct
+    {
+        struct UNISend
+        {
+            u16 state;
+            u8 dataReadyFlag;
+            u8 bmSlot;
+            u16 payloadSize;
+            const void *src;
+        } send;
+        struct UNIRecv
+        {
+            u16 state;
+            u16 errorCode;
+            u16 dataSize;
+            u8 newDataFlag;
+            u8 dataBlockFlag;
+        } recv;
+    } sub;
+    void *recvBuffer;
+    u32 recvBufferSize;
 };
 
-struct RfuUnk2
+struct NIComm
 {
-    u16 unk_0;
-    u16 unk_2;
-    u8 fill_4[0x16];
-    u8 unk_1a;
-    u8 fill_1b[0x19];
-    u16 unk_34;
-    u16 unk_36;
-    u8 fill_38[0x16];
-    u8 unk_4e;
-    u8 fill_4f[0x12];
-    u8 unk_61;
-    u8 fill_62[6];
-    void *unk_68;
-    void *unk_6c;
-    u8 unk_70[0x70];
+    u16 state;
+    u16 failCounter;
+    const u8 *nowP[4];
+    s32 remainSize;
+    u16 errorCode;
+    u8 bmSlot;
+    u8 recvAckFlag[4];
+    u8 ack;
+    u8 phase;
+    u8 n[4];
+    const void *src;
+    u8 bmSlotOrg;
+    u8 dataType;
+    u16 payloadSize;
+    u32 dataSize;
 };
 
-struct RfuUnk3
+struct RfuSlotStatusNI
 {
-    u32 unk_0;
-    u32 unk_4;
-    u8 unk_8[0xD4];
-    u32 unk_dc;
+    struct
+    {
+        struct NIComm send;
+        struct NIComm recv;
+    } sub;
+    void *recvBuffer;
+    u32 recvBufferSize;
 };
 
-struct RfuUnk5Sub
+struct RfuFixed
 {
-    u16 unk_00;
-    u8 unk_02;
-    u16 unk_04;
-    struct UnkLinkRfuStruct_02022B14 unk_06;
-    u8 fill_13[1];
-    u8 playerName[PLAYER_NAME_LENGTH + 1];
+    void (*reqCallback)(u16, u16);
+    void (*fastCopyPtr)(const u8 **, u8 **, s32);
+    u16 fastCopyBuffer[24];
+    u32 fastCopyBuffer2[12];
+    u32 LLFBuffer[29];
+    struct RfuIntrStruct *STWIBuffer;
 };
 
-struct RfuUnk5
+struct RfuStatic
 {
-    u8 unk_00;
-    u8 unk_01;
-    u8 unk_02;
-    u8 unk_03;
-    u8 unk_04;
-    u8 unk_05;
-    u8 unk_06;
-    u8 unk_07;
-    u8 unk_08;
-    u8 filler_09[1];
-    u8 unk_0a[4];
-    u8 filler_0e[6];
-    struct RfuUnk5Sub unk_14[4];
+    u8 flags;
+    u8 NIEndRecvFlag;
+    u8 recvRenewalFlag;
+    u8 commExistFlag;
+    u8 recvErrorFlag;
+    u8 recoveryBmSlot;
+    u8 nowWatchInterval;
+    u8 nullFrameCount;
+    u8 emberCount;
+    u8 SCStartFlag;
+    u8 linkEmergencyFlag[4];
+    u8 lsFixedCount[4];
+    u16 cidBak[4];
+    u16 unk_1a;
+    u16 reqResult;
+    u16 tryPid;
+    u16 watchdogTimer;
+    u32 totalPacketSize;
 };
 
-extern struct RfuStruct * gRfuState;
+struct RfuTgtData
+{
+    u16 id;
+    u8 slot;
+    u8 multibootFlag;
+    u16 serialNum;
+    u8 gname[15];
+    u8 uname[9];
+};
```

Comment:
// Game Identification Information Data Structure
typedef struct rfuTgtDataTag {                             // During SearchParent                                    All other times (Parent or Child)
    u16 id;                                                // ID of parent candidate                                 ID of connection partner
    u8  slot;                                              // Slot number where parent candidate can enter           Connection slot number
    u8  mboot_flag;                                        // Flag to indicate whether or not the parent candidate   Flag to indicate whether or not the connection partner
                                                           //   is multiboot program for download                      is multiboot program for download
    u16 serialNo;                                          // Game serial number of parent candidate                 Game serial number of connection partner
    u8  gname[RFU_GAME_NAME_LENGTH+2];                     // Game name of parent candidate                          Game name of connection partner
    u8  uname[RFU_USER_NAME_LENGTH+1];                     // User name for parent candidate                         User name for connection partner
        u8  padding[2];
}rfuTgtData;